### PR TITLE
LTD-3131: Add view and form to edit trigger list assessment

### PR DESCRIPTION
**Pull request by saruniitr**
## Change description

Add view and form to edit trigger list assignment.

Some templates are used outside of TAU in Advice app hence they are moved to a common location caseworker/core/templates/goods/includes

If these are moved to any other location in future then ensure they are in a templates directory in an app because these used in good_detail.html which is rendered as form widget and we get `TemplateDoesnotExist` error if they are in a directory other than app's templates directory.

### CODE REVIEW

File: caseworker/advice/templates/advice/trigger_list_edit.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    <h1 class="govuk-heading-l govuk-!-margin-top-6">Product assessment</h1>
+    <p id="subtitle" class="govuk-body">Editing assessment for <strong>{{ good_on_application.good.name }}</strong></p>
+    <main class="govuk-main-wrapper">
+        <div class="govuk-grid-row">
+            <div class="govuk-grid-column-two-thirds">
+                <div>
+                    <h3 class="govuk-heading-m">{{ good_on_application.line_number }}. {{ good_on_application.good.name }}</h3>
+                    <div>
+                      {% include "tau/good_table.html" with good=good_on_application summary=summary %}
+                      {% include "tau/precedent_table.html" with precedents=good_on_application.precedents %}
```
Comment (kevincarrogan) [inline caseworker/advice/templates/advice/trigger_list_edit.html:14]: If we're using this outside of just TAU we should put these in a folder that is more generic.

File: caseworker/advice/templates/advice/trigger_list_edit.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                    </div>
+                </div>
+            </div>
+            <div class="govuk-grid-column-one-third">
+                <form id="tau-form" method="POST">
```
Comment (kevincarrogan) [inline caseworker/advice/templates/advice/trigger_list_edit.html:19]: This isn't a `tau-form` so the id should be different here. We actually might not even need the id really.

File: caseworker/advice/templates/advice/trigger_list_edit.html [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+                    {% csrf_token %}
+                    <h1 class="govuk-heading-m tau__headline">Assessing {{ good_on_application.good.name}}</h1>
```
Comment (kevincarrogan) [inline caseworker/advice/templates/advice/trigger_list_edit.html:21]: I think this class is now wrong and shouldn't be `tau__headline`.

File: caseworker/advice/templates/advice/trigger_list_edit.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                </form>
+            </div>
+        </div>
+    </main>
+</div>
+{% endblock %}
+
+{% block javascript %}
+    {{ block.super }}
+{% endblock %}
```
Comment (kevincarrogan) [inline caseworker/advice/templates/advice/trigger_list_edit.html:32]: If this is just including `{{ block.super }}` then we don't need the block at all.

File: caseworker/advice/views.py [hunk 5]
```diff
@@ -574,15 +582,7 @@ def get_success_url(self):
...
-class BEISProductAssessment(AdviceView, BEISNuclearMixin, FormView):
-    """This renders trigger list product assessment for BEIS Nuclear"""
-
-    template_name = "advice/trigger_list_home.html"
-    form_class = BEISTriggerListAssessmentForm
-
-    def get_success_url(self):
-        return reverse("cases:advice_view", kwargs=self.kwargs)
-
+class BEISAssessmentView:
```
Comment (kevincarrogan) [inline caseworker/advice/views.py:585]: This isn't really a `View` so I don't think it should be called this.

File: caseworker/advice/views.py [hunk 6]
```diff
@@ -592,6 +592,38 @@ def organisation_documents(self):
...
+                **good_on_application_map[item_id],
+                **data,
+            }
+            for item_id in selected_good_ids
+        ]
+
+        return services.post_trigger_list_assessment(self.request, case_id=self.kwargs["pk"], data=data)
+
+
+class BEISProductAssessment(AdviceView, BEISNuclearMixin, BEISAssessmentView, FormView):
```
Comment (kevincarrogan) [inline caseworker/advice/views.py:618]: This is actually a view but isn't called `View`.

File: caseworker/advice/templates/advice/trigger_list_edit.html (hunk 4 above)
Comment (kevincarrogan) [reply caseworker/advice/templates/advice/trigger_list_edit.html:32]: Also, for the TAU edit there's a different JS script for this. I think we might need to do the same thing here. Is the JS working on this page? Do you see any errors in the console when you're on this page?

Comment (saruniitr) [reply caseworker/advice/templates/advice/trigger_list_edit.html:32]: will remove this as there is no need for JS in this case

File: caseworker/advice/templates/advice/trigger_list_edit.html (hunk 3 above)
Comment (saruniitr) [reply caseworker/advice/templates/advice/trigger_list_edit.html:21]: removed

File: caseworker/advice/templates/advice/trigger_list_edit.html (hunk 1 above)
Comment (saruniitr) [reply caseworker/advice/templates/advice/trigger_list_edit.html:14]: :+1:

**Review by kevincarrogan — APPROVED** [review]

**Pull request merged**